spock-specs/src/test/groovy/org/spockframework/docs/interaction/DetachedMockFactoryDocSpec.groovy

```diff
@@ -0,0 +1,149 @@
+package org.spockframework.docs.interaction
+
+import org.spockframework.mock.IDefaultResponse
+import org.spockframework.mock.IMockInvocation
+import org.spockframework.mock.MockUtil
+import org.spockframework.mock.ZeroOrNullResponse
+import spock.lang.Specification
+import spock.lang.Unroll
+import spock.mock.AutoAttach
+import spock.mock.DetachedMockFactory
+
+import static org.spockframework.docs.interaction.DetachedMockFactoryDocSpec.EngineMockCreator.StartMode.*
+
+class DetachedMockFactoryDocSpec extends Specification {
+  // tag::declare-static[]
+  static mockFactory = new DetachedMockFactory()
+  static mockUtil = new MockUtil()
+  // end::declare-static[]
+
+  // tag::attach-manually[]
+  def "Manually attach detached mock"() {
+    given:
+    def manuallyAttachedEngine = mockFactory.Mock(Engine)
+    manuallyAttachedEngine.isStarted() >> true
+    mockUtil.attachMock(manuallyAttachedEngine, this)
```

Comment:
Although it works, I think we should show this order, to first attach the mock and then define behavior for it.
```suggestion
    mockUtil.attachMock(manuallyAttachedEngine, this)
    manuallyAttachedEngine.isStarted() >> true
```

Reply:
Absolutely, I agree. Good catch! I will also adjust the explanation in the corresponding docs section.